integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/TopicBasicsTestCase.java

```diff
@@ -0,0 +1,167 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.ei.mb.test.amqp;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+import org.wso2.ei.mb.test.utils.ClientConstants;
+import org.wso2.ei.mb.test.utils.JMSAcknowledgeMode;
+import test.java.org.wso2.ei.mb.test.amqp.BrokerTest;
+import test.java.org.wso2.ei.mb.test.client.TopicReceiver;
+import test.java.org.wso2.ei.mb.test.client.TopicSender;
+
+import java.io.IOException;
+import javax.jms.JMSException;
+import javax.naming.NamingException;
+
+/**
+ * Test cases for the basic functionality the Topics.
+ */
+public class TopicBasicsTestCase extends BrokerTest {
+
+    /**
+     * The logger used in logging information, warnings, errors and etc.
+     */
+    private static final Logger log = Logger.getLogger(TopicBasicsTestCase.class);
+
+    @BeforeClass
+    public void init() throws Exception {
+        super.init();
+    }
+
+    /**
+     * In this test method we check sending messages to a Topic and receiving them.
+     * 1. Register two subscribers to a Topic.
+     * 2. Send a message.
+     * 3. Receive them by two subscribers.
+     * 4. Send 500 messages.
+     * 5. Receive them by two subscribers.
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     */
+    @Test(groups = "wso2.mb.topic")
+    public void performTopicSendReceive() throws JMSException, NamingException, IOException, InterruptedException {
+
+        TopicSender sender = null;
+        TopicReceiver subscriber = null;
+        TopicReceiver subscriber2 = null;
+        try {
+            subscriber = new TopicReceiver("TopicBasicsTestCaseSendReceive", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                    configurationReader, null);
+
+            subscriber2 = new TopicReceiver("TopicBasicsTestCaseSendReceive", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                    configurationReader, null);
+
+            //send a message
+            sender = new TopicSender("TopicBasicsTestCaseSendReceive", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                    configurationReader);
+            sender.sendMessages(1, "test message");
+
+            Assert.assertNotNull(subscriber.receiveMessage(ClientConstants.CLIENT_WAIT_DURATION),
+                    "Topic not received by the subscriber");
+            Assert.assertNotNull(subscriber2.receiveMessage(ClientConstants.CLIENT_WAIT_DURATION),
+                    "Topic not received by the subscriber2");
+
+            //send 500 messages
+            sender.sendMessages(500, "test message");
```

Comment:
Better if we can have a constant for 500, that way we don't need to change it in several places in the case of a modification.

Reply:
added as a variable.